radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/addressbook/AddressBook.java

```diff
@@ -227,20 +194,28 @@ private boolean sameNetworkHrp(RadixNodeUri uri) {
           .filter(not(AddressBookEntry::isBanned))
           .flatMap(e -> e.getKnownAddresses().stream())
           .filter(not(PeerAddressEntry::blacklisted))
-          .sorted(addressEntryComparator)
           .map(AddressBookEntry.PeerAddressEntry::getUri)
+          .filter(not(postponingManager::shouldIgnore))
+          .sorted()
           .findFirst();
     }
   }
 
   public void addOrUpdatePeerWithSuccessfulConnection(RadixNodeUri radixNodeUri) {
     this.addOrUpdatePeerWithLatestConnectionStatus(radixNodeUri, LatestConnectionStatus.SUCCESS);
-    this.addressEntryComparator.resetFailures(radixNodeUri);
+
+    this.postponingManager.recordSuccess(radixNodeUri);
   }
 
   public void addOrUpdatePeerWithFailedConnection(RadixNodeUri radixNodeUri) {
     this.addOrUpdatePeerWithLatestConnectionStatus(radixNodeUri, LatestConnectionStatus.FAILURE);
-    this.addressEntryComparator.incFailures(radixNodeUri);
+
+    if (!this.postponingManager.recordFailure(radixNodeUri)) {
+      synchronized (lock) {
```

Comment:
Is this lock necessary? `knownPeers` is thread-safe. Not sure bout `persistence` tho.

Reply:
Persistence layers is not thread safe

Reply:
would it be too hard to do that?